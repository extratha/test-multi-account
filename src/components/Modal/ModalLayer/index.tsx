'use client';

import { Container, Modal, Paper, styled } from '@mui/material';

import useModal from '@/store/modal';

const StyledModal = styled(Modal)(() => ({
  position: 'fixed',
  zIndex: 999,
  right: 0,
  bottom: 0,
  top: 0,
  left: 0,
  display: 'flex',
  alignItems: 'center',
  justifyContent: 'center',
  outline: 0,
  '& :focus-visible': {
    outline: 0,
  },
}));

const StyledPaper = styled(Paper)(() => ({
  padding: '24px',
  margin: '0 16px',
  borderRadius: 28,
  display: 'flex',
  alignItems: 'center',
  justifyContent: 'center',
  flexDirection: 'column',
  textAlign: 'center',
}));

const ModalLayer = () => {
  const { modal, closeModal } = useModal();
  const ModalContent = modal.content;

  return (
    <StyledModal disableEnforceFocus open={modal.isVisible}>
      <Container maxWidth="sm" disableGutters>
        <StyledPaper elevation={3}>
          <ModalContent closeModal={closeModal} />
        </StyledPaper>
      </Container>
    </StyledModal>
  );
};

export default ModalLayer;
