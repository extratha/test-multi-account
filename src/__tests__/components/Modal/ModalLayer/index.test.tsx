import ModalLayer from "@/components/Modal/ModalLayer";
import useModal from "@/store/modal";
import { act, render, renderHook, screen } from "@testing-library/react";

jest.mock("../../../../store/modal", () => {
  return {
    __esModule: true,
    default: jest.fn(() => ({
      modal: { isVisible: false, content: () => null },
      openModal: jest.fn(),
      closeModal: jest.fn(),
    })),
  };
});

describe("ModalLayer component", () => {
  const closeModalMock = jest.fn();
  const openModalMock = jest.fn();

  beforeEach(() => {
    // jest.clearAllMocks();
  });

  const mockModalContent = () => (
    <div data-testid="modal-content">
      Modal Content
      <button onClick={closeModalMock}>Close</button>
    </div>
  );

  it("renders the modal when visible", () => {
    (useModal as any).mockReturnValue({
      modal: {
        isVisible: true,
        content: mockModalContent,
      },
      closeModal: closeModalMock,
      openModal: openModalMock,
    });
    openModalMock();
    closeModalMock();
    render(<ModalLayer />);
    expect(screen.getByTestId("modal-content")).toBeInTheDocument();
    expect(screen.getByText("Modal Content")).toBeInTheDocument();
  });

  it("opens the modal with the correct content", () => {
    const { result } = renderHook(() => useModal());

    act(() => {
      result.current.openModal(mockModalContent);
      result.current.modal.isVisible = true;
    });

    const { asFragment } = render(<ModalLayer />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("closes the modal", async () => {
    const { result } = renderHook(() => useModal());

    act(() => {
      result.current.openModal(mockModalContent);
      result.current.modal.isVisible = false;
    });

    const { asFragment } = render(<ModalLayer />);
    expect(asFragment()).toMatchSnapshot();
  });
});
