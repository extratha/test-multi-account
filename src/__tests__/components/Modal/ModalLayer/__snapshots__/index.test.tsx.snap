// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalLayer component closes the modal 1`] = `<DocumentFragment />`;

exports[`ModalLayer component opens the modal with the correct content 1`] = `<DocumentFragment />`;
